### DISCUSSION TITLE
Retirement of rgdal, rgeos, and maptools

### DISCUSSION THREAD

**Discussion by tkrisztin** [iiasa/R_collaboration#3] [General]
See the article here:

[https://www.r-bloggers.com/2022/04/r-spatial-evolution-retirement-of-rgdal-rgeos-and-maptools/](https://www.r-bloggers.com/2022/04/r-spatial-evolution-retirement-of-rgdal-rgeos-and-maptools/)

As many of us are using these packages, and the **raster** package is especially impacted, it would be good to discuss strategies on how to change code for the **terra** package and other recommended changes,

**Comment by Martin-Jung** [comment]
We have tried to make the switch to `terra` in the past (Dec 2021), however soon defaulted back to `raster` as there (a) were still numerous bugs and memory leaks with `terra`, (b) not all functionalities of the `raster` package have yet been ported.
Would be interested to hear experiences from others.

We don't really use `maptools` and I think most functionalities that this package offers can easily be replaced by simple features (`sf` package). See also the [spatial analysis course](https://github.com/iiasa/R_collaboration/wiki/Spatial-data-and-analysis) for ESM that I created a while ago.

Similar to `terra` I would also encourage people to check out `stars`. I have been using it since a while and it offers excellent capabilities for handling spatial-temporal gridded data (opposed to `raster` and `terra`). Loading and analysing a netcdf file with multiple dimensions and attributes is relatively (with a bit of pain ;-) ) straight forward.
[https://r-spatial.github.io/stars/](https://r-spatial.github.io/stars/)

**Reply by michidoubleu** [reply]
I have to agree with Martin, I tried the `terra` package, but had troubles to migrate the whole code due to lack of some functions available in `raster`. Nonetheless, for the task of aggregating high-res raster information to SimUs or NUTS regions `terra` and `stars` paired with a package called `exactextractr` led to a nice solution, while with the `raster` I ran into troubles. 

So for me it currently depends on the task, if I stick to `raster` or already employ the newer packages and I will wait until they enter a more mature state unitl I migrate completely.

**Comment by michielvandijk** [comment]
Greetings from the Netherlands! Actually, I am currently in the process of transitioning to the terra package. So far my experience is that it is MUCH better able to handle large spatial files and implement operations on such files, such as recoding, resampling, etc due to the internal GDAL bindings. This is completely impossible with raster. Before I combined exactextractr with raster for aggregating over polygons, which indeed works great. I have not tested the extract function in terra but I assume this also must be much better now. For complex operations I used gdalUtilities and sometimes GDAL directly but the latter is a bit of a pain as I do not like the command line. In sum, so far my experience with terra was very positive but I am still exploring so maybe this will change.

**Reply by Martin-Jung** [reply]
Do make consistency checks. We have used `terra` for extractions end of last year and found that results were inconsistent with what raster would extract. It also occassionaly breaks the code as @matthewlewis896 found.

Overall `terra` is great, but I think it needs a bit more development / code maturity before it is a stand-in replacement
